### PR TITLE
junit5 dynamic tests from TableData

### PR DESCRIPTION
**Pull request by MykolaGolubyev**
```
class DynamicTestsGroovyTest {
    @TestFactory
    def priceCalculator() {
        ["price", "quantity", "outcome"] {
        _________________________________
          10     |  30      |  300
          -10    |  30      | -300
        }.useCases {
            PriceCalculator.calculate(price, quantity).should == outcome
        }
    }
}
```

### CODE REVIEW

File: webtau-junit5/src/main/java/com/twosigma/webtau/junit5/DynamicTests.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+public class DynamicTests {
+    private DynamicTests() {
+    }
+
+    public static Stream<DynamicTest> fromTable(String useCasesPrefix, TableData useCases, Consumer<Record> test) {
+        String displayNamePrefix = useCasesPrefix.isEmpty() ? "" : useCasesPrefix + ": ";
```
Comment (tsiq-clemens) [inline webtau-junit5/src/main/java/com/twosigma/webtau/junit5/DynamicTests.java]: should it be empty or the row number? I think the latter is the DynamicTest default for java.

Comment (tsiq-clemens) [reply webtau-junit5/src/main/java/com/twosigma/webtau/junit5/DynamicTests.java]: or maybe I misunderstand the display name approach (I'll take it for a spin). How about optionally designating a column as display name? If you have more than 2 columns or lengthier values, the names can easily get too long for meaningful display.

**Comment by MykolaGolubyev** [comment]
@tsiq-karold, @tsiq-clemens having doubts about `useCases` function name. Should it be `forEachUseCase` or something like that?

File: webtau-junit5/src/main/java/com/twosigma/webtau/junit5/DynamicTests.java (hunk 1 above)
Comment (MykolaGolubyev) [reply webtau-junit5/src/main/java/com/twosigma/webtau/junit5/DynamicTests.java]: display name for each case will be `Record.toString()`. Prefix is optional text to put in front. 
I thought about `display` like column but then ran tests from IDEA and it was clear to me what is going on. Also if you want a row to have a dedicated test name, maybe it should be its own test method?

**Comment by tsiq-karold** [comment]
`forEachTestCase`?

**Comment by MykolaGolubyev** [comment]
The only confusion part I see with `forEach...` is that `forEach` semantic is not to return anything. Current `useCases` actually returns a `Stream<DynamicTest>`. 
Should it then be `mapTestCase`?

**Comment by tsiq-karold** [comment]
Hmm, that's fair.  How about `testCode`?

**Comment by MykolaGolubyev** [comment]
`test`?

**Comment by tsiq-karold** [comment]
I'm happy with that.

File: webtau-junit5/src/main/java/com/twosigma/webtau/junit5/DynamicTests.java (hunk 1 above)
Comment (tsiq-clemens) [reply webtau-junit5/src/main/java/com/twosigma/webtau/junit5/DynamicTests.java]: As discussed, we'll add an optional "label" column feature in a separate PR

**Comment by MykolaGolubyev** [comment]
`.validate` ?

**Comment by tsiq-clemens** [comment]
or `.check` ?

**Comment by MykolaGolubyev** [comment]
OK, settling down on `test`. Maybe we will add aliases in future.

**Pull request merged**